pkg/cmd/server/api/v1/conversions.go

```diff
@@ -307,6 +307,23 @@ func addConversionFuncs(scheme *runtime.Scheme) {
 			out.Location = in.Location
 			return nil
 		},
+		func(in *MasterVolumeConfig, out *internal.MasterVolumeConfig, s conversion.Scope) error {
+			if in.DynamicProvisioningEnabled != nil && !*in.DynamicProvisioningEnabled {
+				out.DynamicProvisioningEnabled = false
+				return nil
+			}
+			out.DynamicProvisioningEnabled = true
+			return nil
+		},
+		func(in *internal.MasterVolumeConfig, out *MasterVolumeConfig, s conversion.Scope) error {
```

Comment:
if we want new configs to default to true, we need to set this to true in `BuildSerializeableMasterConfig`, or the empty state internal field will convert to an explicit "false" in the serialized config